internal/server/errors.go

```diff
@@ -77,6 +75,20 @@ func (a *API) sendAPIError(c *gin.Context, err error) {
 	c.Abort()
 }
 
+func trimErrorMsg(errBase error, errFull error) string {
+	base, full := errBase.Error(), errFull.Error()
+
+	switch {
+	case base == full:
+		return ""
+	case strings.HasPrefix(full, base+": "):
+		return strings.TrimPrefix(full, base+": ")
+	case strings.HasSuffix(full, ": "+base):
+		return strings.TrimSuffix(full, ": "+base)
+	}
+	return full
+}
```

Comment:
This is not ideal for sure. I think if we used domain-specific errors we wouldn't have this problem. `validator.ValidationErrors` in `sendAPIError`  is a good example of this. We don't have to filter out a message for that error, because the error text is specific to what failed, not a generic mirror of HTTP status codes.

Reply:
This is trimming things like `bad request: ...`? It's not super clear to me.

Reply:
Yup, exactly! I think the new test cases below demonstrate it, but definitely it's not super obvious from the code. I'll add a godoc string to clarify.